### ISSUE TITLE
FNX3-16731 ⁃ [Bug] Sudden crash while typing in the URL bar

### ISSUE THREAD

**Issue by sv-ohorvath** [mozilla-mobile/fenix#3003]
## Steps to reproduce
1. Tap the search bar an type in something. (can't reproduce this with the same keywords)
    1. Expected behavior
URL/search term is typed
    1. Actual behavior
App crashes.
[logcat](https://github.com/mozilla-mobile/fenix/files/3237472/logcat.txt)

### Device information
* Android device: Pixel (Android 9)
* Fenix version: 5/30 build

**Comment by AaronMT** [comment]
```java
2019-05-30 17:38:32.887 24288-24288/? E/AndroidRuntime: FATAL EXCEPTION: main
    Process: org.mozilla.fenix, PID: 24288
    java.lang.NullPointerException: Attempt to invoke interface method 'boolean java.util.Collection.contains(java.lang.Object)' on a null object reference
        at androidx.collection.MapCollections.retainAllHelper(MapCollections.java:4)
        at androidx.fragment.app.FragmentTransition.captureOutSharedElements(FragmentTransition.java:11)
        at androidx.fragment.app.FragmentTransition.startTransitions(FragmentTransition.java:51)
        at androidx.fragment.app.FragmentManagerImpl.executeOpsTogether(FragmentManagerImpl.java:81)
        at androidx.fragment.app.FragmentManagerImpl.removeRedundantOperationsAndExecute(FragmentManagerImpl.java:11)
        at androidx.fragment.app.FragmentManagerImpl.execPendingActions(FragmentManagerImpl.java:4)
        at androidx.fragment.app.FragmentManagerImpl$2.run(FragmentManagerImpl.java:1)
        at android.os.Handler.handleCallback(Handler.java:873)
        at android.os.Handler.dispatchMessage(Handler.java:99)
        at android.os.Looper.loop(Looper.java:193)
        at android.app.ActivityThread.main(ActivityThread.java:6718)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858)
2019-05-30 17:38:32.887 24288-24288/? I/mozac/CrashReporter: Received crash: UncaughtExceptionCrash(throwable=java.lang.NullPointerException: Attempt to invoke interface method 'boolean java.util.Collection.contains(java.lang.Object)' on a null object reference)

**Comment by softvision-miralobontiu** [comment]
Verified as fixed with the latest build 1.0.1922 (#11510609), with Motorola Moto G6 (Android 8.0), Sony Xperia Z5 Premium (Android 7.1.1), and Samsung Galaxy Tab A6 (Android 5.1.1).
